### ISSUE TITLE
filesystem::last_write_time must require an explicit replica number or update all replicas

### ISSUE THREAD

**Issue by korydraughn** [irods/irods#5061]
- [x] master
- [x] 4-2-stable
---
## Bug Report
`irods::filesystem::last_write_time` does not allow you to update a specific replica. It also doesn't update all replicas. It will update whatever replica is picked by the database plugin. The API should not allow this.

## Solution
Adjust the existing API and add an additional function that supports targeting a specific replica.
```c++
// Updates the mtime of ALL replicas.
auto last_write_time(conn, path, new_mtime) -> void;

// Updates the mtime of the replica identified by replica_number.
auto last_write_time(conn, path, replica_number, new_mtime) -> void;
```

**Comment by korydraughn** [comment]
One of the original requirements of the  filesystem API was that it only deal with the logical layer. The filesystem API should NOT handle replicas in any way. Replica operations should be handled through a different library and work with the filesystem API in a seamless way to help developers reach a solution.

**Comment by korydraughn** [comment]
This will be resolved by #5103

**Comment by korydraughn** [comment]
Resolved by #5130 and #5136
